### PR TITLE
fixed issue 251 Do not use globals that need to be reset in tests

### PR DESCRIPTION
**Pull request by benmoore422**
Fixed issue 251 Do not use globals that need to be reset in tests

Removed unused variable `on_recv_cbs` as well as unused function `track_recv`. Also changed network into contructor argument for `DummyNetwork` class in `raiden/raiden/network/transport.py` 

Moved transport instantiation out of `App` constructor in `raiden/raiden/app.py`

### CODE REVIEW

**Comment by himynameisbeni** [comment]
I'm not sure why the build is failing I've looked into it and it seems to be failing on an issue not with my commit but I could be wrong. If anyone has any insight I'd be happy to know what I did wrong.

**Comment by hackaugusto** [comment]
Most of the errors are related to the linter, we use flake8 to enforce the pep8 style.

You can install git hooks to run the linter locally, the commit will fail if the new code break the style guideles instead of waiting for the travis build:

http://flake8.pycqa.org/en/latest/user/using-hooks.html

**Comment by LefterisJP** [comment]
Also @himynameisbeni please squash all your commits into one when ready. We want the git history to be clean.

File: raiden/network/transport.py [hunk 1]
```diff
@@ -132,11 +132,10 @@ class DummyNetwork(object):
     network protocol just greenlet communication.
     """
 
-    on_send_cbs = []  # debugging
-
-    def __init__(self):
+    def __init__(self, on_send_cbs=[]):
```
Comment (hackaugusto) [inline raiden/network/transport.py]: Note that mutable defaults are still global values, e.g.:

```python
>>> def a(l=[]):
...   l.append(1)
...   print(l)
...
>>> a()
[1]
>>> a()
[1, 1]
>>> a()
[1, 1, 1]
```

File: raiden/app.py [hunk 2]
```diff
@@ -50,22 +50,11 @@ class App(object):  # pylint: disable=too-few-public-methods
         'shutdown_timeout': DEFAULT_SHUTDOWN_TIMEOUT,
     }
 
-    def __init__(self, config, chain, default_registry, discovery, transport_class=UDPTransport):
+    def __init__(self, config, chain, default_registry, discovery, transport=UDPTransport):
+        transport = self.transport_instantiation(self, config, transport)
```
Comment (hackaugusto) [inline raiden/app.py]: You will probably need to move this out of the constructor, as in `pass it as a constructor argument` or `inject the transport through the constructor`, e.g.:

```python
def __init__(self, config, chain, default_registry, discovery, transport):
   self.transport = transport
   ...
```

Comment (himynameisbeni) [reply raiden/app.py]: Ah I see, thanks I will give that a try. I really appreciate the help! Sorry if I'm clogging up the commit history I will squash it all when I'm done.

**Comment by palango** [comment]
Closing this as there wasn't any activity recently. Feel free to open again when you continue working on this issue.

**Pull request closed without merging**